packages/stack/src/views/Stack/CardContainer.tsx

```diff
@@ -109,6 +115,9 @@ function CardContainer({
   React.useEffect(() => {
     onPageChangeConfirm?.();
   }, [active, onPageChangeConfirm]);
+  React.useEffect(() => {
+    onGestureEnd?.({ route: scene.route });
```

Comment:
`onGestureEnd` needs to be called when gesture ends, but here seems like it's called if `scene`, `active` or `onGestureEnd` props change + on mount. I think you need to add a separate `onGestureEnd` callback in `Card` to handle it.

Reply:
Ah, good catch. Yeah i did actually add the callback in `Card.tsx`, but for some reason got confused here. I've made the change